src/irmin/tree.ml

```diff
@@ -211,22 +230,25 @@ module Make (P : Private.S) = struct
           v
       | _, v -> v
 
-    let hash c =
-      match cached_hash c with
+    let hash t =
+      match cached_hash t with
       | Some k -> k
       | None -> (
-          match cached_value c with
+          match cached_value t with
           | None -> assert false
           | Some v ->
               cnt.contents_hash <- cnt.contents_hash + 1;
-              let k = P.Contents.Key.hash v in
-              c.info.hash <- Some k;
-              k)
+              let h = P.Contents.Hash.hash v in
+              assert (t.info.ptr = Ptr_none);
+              t.info.ptr <- Hash h;
```

Comment:
this contradicts your comment above about caching `Key` over `Hash`, no?

Reply:
I believe we have no `Key` available to cache here.

Given just the value, we can't compute its key, but we can compute its hash.